chickadee/processes/wps_ca.py

```diff
@@ -0,0 +1,177 @@
+import os
+from pywps import Process, ComplexOutput, LiteralInput, FORMATS
+from pywps.app.Common import Metadata
+from netCDF4 import Dataset
+from rpy2 import robjects
+
+from wps_tools.utils import log_handler
+from wps_tools.io import log_level
+from chickadee.utils import logger, set_r_options, get_package
+from chickadee.io import gcm_file, obs_file, varname, num_cores, end_date
+
+
+class CA(Process):
+    """Constructed Analogue (CA) downscaling algorithm:
+    Starts by spatially aggregating high-resolution gridded observations
+    up to the scale of a GCM. Then it proceeds to bias correcting the GCM
+    based on those observations. Finally, it conducts the search for temporal
+    analogues. This involves taking each timestep in the GCM and searching for
+    the top 30 closest timesteps in the gridded observations. For each of the
+    30 closest "analogue" timesteps, CA records the integer number of the
+    timestep (indices) and a weight for each of the analogues."""
+
+    def __init__(self):
+        self.status_percentage_steps = {
+            "start": 0,
+            "process": 10,
+            "write_files": 80,
+            "build_output": 95,
+            "complete": 100,
+        }
+
+        inputs = [
+            gcm_file,
+            obs_file,
+            varname,
+            num_cores,
+            end_date,
+            LiteralInput(
+                "indices",
+                "Indices File",
+                abstract="File name to store indices of analogue times steps (suffix .txt)",
+                data_type="string",
+            ),
+            LiteralInput(
+                "weights",
+                "Weights File",
+                abstract="File name to store weights of analogues (suffix .txt)",
+                data_type="string",
+            ),
```

Comment:
I think it might work, but I was having trouble with it. I was making a file in the notebook to use, but then I was unable to access it on dev-03. I played around with it for a bit, and couldn't get it to work. I'm very happy for suggestions if you or @sum1lim have any, as you have more experience with the processes. I can also poke at it for longer.